lib/mayaUsd/fileio/jobs/jobArgs.cpp

```diff
@@ -487,81 +597,55 @@ bool _MergeJobContexts(bool isExport, const VtDictionary& userArgs, VtDictionary
-                    if (v.IsHolding<std::vector<VtValue>>()) {
-                        // We merge arrays:
-                        std::vector<VtValue> mergedValues
-                            = allContextValue.UncheckedGet<std::vector<VtValue>>();
-                        for (const VtValue& element : v.UncheckedGet<std::vector<VtValue>>()) {
-                            if (element.IsHolding<std::vector<VtValue>>()) {
-                                // vector<vector<string>> is common for chaserArgs and shadingModes
-                                auto findElement = [&element](const VtValue& a) {
-                                    return UsdUfe::compareValues(element, a)
-                                        == UsdUfe::DiffResult::Same;
-                                };
-                                if (std::find_if(
-                                        mergedValues.begin(), mergedValues.end(), findElement)
-                                    == mergedValues.end()) {
-                                    mergedValues.push_back(element);
-                                }
-                            } else {
-                                if (std::find(mergedValues.begin(), mergedValues.end(), element)
-                                    == mergedValues.end()) {
-                                    mergedValues.push_back(element);
-                                }
-                            }
-                        }
-                        allContextArgs[k] = mergedValues;
-                    } else {
-                        // We have both an array and a scalar under the same argument name.
-                        TF_RUNTIME_ERROR(TfStringPrintf(
-                            "Context '%s' and context '%s' do not agree on type of argument '%s'.",
-                            sourceName.c_str(),
-                            argInitialSource[k].c_str(),
-                            k.c_str()));
-                        canMergeContexts = false;
-                    }
-                } else {
-                    // Scalar value already exists. Check for value conflicts:
-                    if (allContextValue != v) {
-                        TF_RUNTIME_ERROR(TfStringPrintf(
-                            "Context '%s' and context '%s' do not agree on argument '%s'.",
-                            sourceName.c_str(),
-                            argInitialSource[k].c_str(),
-                            k.c_str()));
-                        canMergeContexts = false;
-                    }
-                }
-            }
-        }
+
+        const bool allowConflicts = false;
+        if (!_MergeDictionaries(allContextArgs, argInitialSource, dict, sourceName, allowConflicts))
+            canMergeContexts = false;
     }
     return canMergeContexts;
 }
 
+VtDictionary
+_MergeAllArguments(bool isExport, const VtDictionary& userArgs, const VtDictionary& defaultsArgs)
+{
```

Comment:
This function is refactored from the import/export job args constructors found below, but now we properly merge user arg and job option args instead of overwriting users args entirely.